contracts/src/ethereum/TokenBridge.sol

```diff
@@ -140,27 +144,48 @@ contract TokenBridge is
         l1Token.transfer(recipient, amount);
     }
 
-     function consumeBridgeRewardMessage(address l1Token, address recipient, uint256 amount) internal {
-        emit LogBridgeReward(l1Token, recipient, amount);
+     function consumeBridgeRewardMessage(address recipient, uint256 amount) internal {
+        emit LogBridgeReward(recipient, amount);
 
-        uint256[] memory payload = new uint256[](5);
+        uint256[] memory payload = new uint256[](4);
         payload[0] = BRIDGE_REWARD_MESSAGE;
-        payload[1] = uint256(l1Token);
-        payload[2] = uint256(recipient);
-        payload[3] = amount & (UINT256_PART_SIZE - 1);
-        payload[4] = amount >> UINT256_PART_SIZE_BITS;
+        payload[1] = uint256(recipient);
+        payload[2] = amount & (UINT256_PART_SIZE - 1);
+        payload[3] = amount >> UINT256_PART_SIZE_BITS;
 
         messagingContract.consumeMessageFromL2(l2TokenBridge, payload);
     }
 
-    function receiveRewards(address l1Token, address recipient, uint256 amount) isApprovedToken(l1Token) external {
-        consumeBridgeRewardMessage(l1Token, recipient, amount);
+    function receiveRewards(address recipient, uint256 amount) external {
+        consumeBridgeRewardMessage(recipient, amount);
         require(recipient != address(0x0), "INVALID_RECIPIENT");
```

Comment:
Not always. Ideologically we should allow people to do this

Reply:
I don't think we should burn from a UX perspective, starkgate also makes the same choice.

Reply:
If the user wants to burn then they can transfer the token to themselves and burn it